hashicorp/waypoint
# Add `UI_GetPipelineRun`

## Why the change?

This API will feed the pipeline run detail page.

## What’s the plan?

- [x] Declare rpc and messages
- [x] Implement handlers

### Any deferred tasks?

- [ ] Add more-efficient state functions

## What does it look like?

Given a run of a valid pipeline ([something like this](https://github.com/jgwhite/waypoint-demos/blob/main/pipelines/waypoint.hcl#L30-L44)), running the command below should produce a big ol’ pile of JSON that looks something like [this gist](https://gist.github.com/jgwhite/b2c8d8b22ec10761633c4d8687fdf577).

```sh
./contrib/waypoint-grpc/waypoint-grpc.sh UI_GetPipelineRun '
{
  "pipeline": {
    "id": "01H06BBZH5BK3M7761PD8WP892"
  },
  "sequence": 2
}
'
```

## How do I test it?

1. Run a pipeline
2. From your waypoint source directory, run the following:
   ```sh
   ./contrib/waypoint-grpc/waypoint-grpc.sh UI_GetPipelineRun '
   {
     "pipeline": {
       "id": "<YOUR-PIPELINE-ID>"
     },
     "sequence": <YOUR-RUN-SEQUENCE>
   }
   '
   ```
3. Verify that you see some lovely deeply-nested JSON
